benchexec/tablegenerator/statisticstex.py

```diff
@@ -0,0 +1,295 @@
+    Provides all LatexCommands for a given run_set + stat_list combination
+
+    Args:
+        run_set: A RunSetResult object
+        stat_list: List of ColumnStatistics for each column in run_set
+        current_command: LatexCommand with benchmark_name and displayName already filled
+
+    Yields:
+        All LatexCommands from the run_set + stat_list combination
+    """
+    # Selects the column name
+    def select_column_name(col):
+        return col.display_title if col.display_title else col.title
+
+    # Check for duplicated column name
+    used_column_titles = create_duplication_dictionary(
+        run_set.columns, select_column_name
+    )
+
+    for column, column_stats in zip(run_set.columns, stat_list):
+        column_title = select_column_name(column)
+        duplication_holder = used_column_titles[column_title]
+        duplication_holder.used += 1
+
+        if duplication_holder.count > 1:
+            suffix = util.number_to_roman_string(duplication_holder.used)
+            logging.warning(
+                'Duplicated formatted column name "%s" detected! '
+                "Column names must be unique for Latex. "
+                "Adding suffix %s to column for now",
+                column_title,
+                suffix,
+            )
+            column_title += suffix
+
+        current_command.set_command_part("column_title", column_title)
+
+        yield from _column_statistic_to_latex_command(
+            current_command, column_stats, column
+        )
+
+
+def _column_statistic_to_latex_command(
+    init_command: LatexCommand,
+    column_statistic: ColumnStatistics,
+    column: Column,
+) -> Iterable[LatexCommand]:
+    """Parses a ColumnStatistics to Latex Commands and yields them
+
+    The provided LatexCommand must have specified benchmark_name, display_name and column_name.
+
+    Args:
+        init_command: LatexCommand with not empty benchmark_name and display_name
+        column_statistic: ColumnStatistics to convert to LatexCommand
+        column: Current column with meta-data
+    Yields:
+        A completely filled LatexCommand
+    """
+    if not column_statistic:
+        return
+
+    stat_value: StatValue
+    for stat_name, stat_value in column_statistic.__dict__.items():
+        if stat_value is None:
+            continue
+
+        # Copy command to prevent using filled command parts from previous iterations
+        command = copy.deepcopy(init_command)
+
+        column_parts = stat_name.split("_")
+        if len(column_parts) < 2:
+            column_parts.append("")
+
+        # Some colum_categories use _ in their names, that's why the column_category is the
+        # whole split list except the last word
+        command.set_command_part(
+            "column_category",
+            "".join(
+                util.cap_first_letter(column_part) for column_part in column_parts[0:-1]
+            ),
+        ).set_command_part("column_subcategory", column_parts[-1])
```

Comment:
Due to how the line breaking looks in Python style and due to the fact that here is never a possibility to chain more than two calls in one place, I think that the chaining here makes the code less readable than simply repeating `command` in a few more places.

The combination of chaining with `yield` is also not really nice because it puts the actions in the wrong order (`yield` happens last but is written first).

So please remove chaining for `set_command_part` and `set_command_value`.